apps/microtvm/arduino/template_project/microtvm_api_server.py

```diff
@@ -368,43 +436,33 @@ def _get_platform_version(self, arduino_cli_path: str) -> float:
         return version.parse(str_version)
 
     # This will only be run for build and upload
-    def _check_platform_version(self, options):
+    def _check_platform_version(self, arduino_cli_cmd: str, warning_as_error: bool):
```

Comment:
I don't think this typing is right - `arduino_cli_cmd` could be `str` or `None`. IMO it would be cleaner to convert it to a string in a high-level function, and skip calling `_get_arduino_cli_cmd` here.

Reply:
what are your thoughts here?

Reply:
that makes sense, I moved `_get_arduino_cli_cmd` call in the high-level function